### PR TITLE
Added title case to various heading, titles, etc...

### LINKED ISSUE #2438: Create Title Case component and use for all Nav Items, Section Titles, and Table Headers
Create a wrapper component that changes text to title case using a plugin like this https://github.com/blakeembrey/change-case/tree/master/packages/title-case

Then wrap all the Nav Items, Section Titles, and Table Headers in that component.

Original content ticket #2371

### This task is done when...
 - [x] install title-case in web
 - [x] web/src/components/Instructions.js - wrap what is being set to heading in titleCase
 - [x] web/src/components/Section.js - wrap what is being set to title in titleCase
 - [x] web/src/containers/ApdStateProfileMedicaidOffice.js - wrap {t(`<whatever>.title`)} in titleCase
 - [x] web/src/containers/AssurancesAndCompliances.js - wrap {t(`assurancesAndCompliance.headings.${name}`)} in titleCase
 - [x] web/src/containers/ExecutiveSummaryBudget.js - wrap th text in titleCase
 - [x] web/src/containers/PreviousActivities - wrap {t('previousActivities.outline.instruction.label')} in titleCase
 - [x] web/src/containers/activity/CostAllocateFFP.js - wrap {t(
                  'activities.costAllocate.ffp.quarterlyFFPInstruction.heading'
                )} in titleCase
 - [x] web/src/containers/activity/CostAllocateFFPQuarterly.js - wrap th text in titleCase
 - [x] web/src/containers/activity/Costs.js - wrap h tags in titleCase
 - [x] web/src/containers/activity/EntryDetails.js - wrap return from makeTitle in titleCase
 - [x] web/src/containers/activity/CostAllocation/OtherFunding.js - wrap h tag in titleCase
 - [x] web/src/containers/ApdKeyPerson/ApdKeyPersonForm.js - wrap h tag in titleCase
 - [x] web/src/containers/viewOnly/AssurancesAndCompliances.js - wrap h tag in titleCase
 - [x] web/src/containers/viewOnly/ExecutiveSummary.js - wrap Review.heading in titleCase
 - [x] web/src/reducers/nav.items.js - wrap all labels in titleCase
 - [x] web/src/components/ContinuePreviewButtons.js - wrap return from buildLabel in titleCase

### PR DESCRIPTION
**Pull request by mirano-darren**
resolves #2438 

**Description-**
Added title case to various titles, labels, headings, etc...

**This pull request changes...**

- web/src/components/Instructions.js
- web/src/components/Section.js
- web/src/containers/ApdStateProfileMedicaidOffice.js
- web/src/containers/AssurancesAndCompliances.js
- web/src/containers/ExecutiveSummaryBudget.js
- web/src/containers/PreviousActivities
- web/src/containers/activity/CostAllocateFFP.js
- web/src/containers/activity/CostAllocateFFPQuarterly.js
- web/src/containers/activity/Costs.js
- web/src/containers/activity/EntryDetails.js
- web/src/containers/activity/CostAllocation/OtherFunding.js
- web/src/containers/ApdKeyPerson/ApdKeyPersonForm.js
- web/src/containers/viewOnly/AssurancesAndCompliances.js
- web/src/containers/viewOnly/ExecutiveSummary.js
- web/src/reducers/nav.items.js
- web/src/components/ContinuePreviewButtons.js

**This pull request also touches…**
- web/package.json
- web/package-lock.json

**possible side effects of this change**
 - Could crash the website if titleCase() was called wrong, but I made sure this wasn't the case before creating this PR

**This pull request was tested in the follow ways…**
- Going to each page on the website and making sure it doesn't crash

**Steps to manually verify this change...**

1. See if everything desired is title cased correctly

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [x] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [x] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [x] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/components/Instruction.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ Heading.defaultProps = {
 };
 
 const Instruction = ({ args, reverse, source, headingDisplay, labelFor }) => {
-  const heading = t([source, 'heading'], { defaultValue: false, ...args });
+  const heading = t([source, titleCase('heading')], { defaultValue: false, ...args });
```
Comment (thetif) [inline web/src/components/Instruction.js:32]: you need to put titleCase around the t() function

File: web/src/components/Section.js [hunk 2]
```diff
@@ -4,9 +4,10 @@ import React from 'react';
 import Instruction from './Instruction';
 import { t } from '../i18n';
 import SecondaryNav from './SecondaryNav';
+import { titleCase } from "title-case";
 
 const Section = ({ children, id, resource }) => {
-  const title = t([resource, 'title'], { defaultValue: false });
+  const title = t([resource, titleCase('title')], { defaultValue: false });
```
Comment (thetif) [inline web/src/components/Section.js:10]: you need to put titleCase around the t() function

**Review by thetif — CHANGES_REQUESTED** [review]

**Review by thetif — APPROVED** [review]
Looks good!

**Comment by thetif** [comment]
Sorry @mirano-darren I pulled this down to look at the package-lock file and then ended up pulling Bill's branch in here by mistake and then I couldn't roll it back 🤦. I got it back to how you had it but I lost your git commit history. You still get full credit for this one though!

**Comment by mirano-darren** [comment]
> Sorry @mirano-darren I pulled this down to look at the package-lock file and then ended up pulling Bill's branch in here by mistake and then I couldn't roll it back 🤦. I got it back to how you had it but I lost your git commit history. You still get full credit for this one though!

All good! :D

**Comment by beparticular** [comment]
@mirano-darren @thetif I can see where most of these have been updated. Some I'm not sure, they appear to be title case already in staging. 

One thing I did want to check on, in Assurances and Compliances a couple of the links appear to be missing in this test environment. Can you take a look and see if that's related to something else, or if something happened there? Thanks!

**Comment by beparticular** [comment]
@mirano-darren @thetif 
This is looking good, I can see where most of the changes are. 

There are still some items in the executive summary that look like they should be title case. 
Total cost: 
Start date - End date: Date not specified - Date not specified
Total cost of activity:
Total funding request:  

Those look like they could be the same style as the summaries on Private Contractor Cost or Outcomes and Metrics?

**Review by beparticular — APPROVED** [review]

**Pull request merged**